include/motor-control/core/motion_group.hpp

```diff
@@ -1,2 +1,98 @@
+#pragma once
 
-class MotionGroupManager {};
\ No newline at end of file
+#include <array>
+#include <numeric>
+#include <type_traits>
+#include <variant>
+
+#include "can/core/messages.hpp"
+
+namespace move_group {
+
+template <std::size_t MaxElements>
+class MoveGroup {
+  public:
+    using MoveTypes =
+        std::variant<std::monostate, can_messages::AddLinearMoveRequest>;
+
+    MoveGroup() {}
+
+    /**
+     *
+     * @param move
```

Comment:
we should probably either get rid of these comments or fill them oiut the rest of the way